src/controller/org.controller/org.controller.js

```diff
@@ -257,20 +426,112 @@ async function createUser (req, res) {
   }
 
-  if (newUser.cna_short_name === undefined) {
-    newUser.cna_short_name = cnaShortName
+  Object.keys(req.query).forEach(key => {
+    if (!returned) {
+      if (key === 'new_username') {
+        newUsername = req.query.new_username.replace(/["']/g, '')
+        newUser.username = newUsername
+      } else if (key === 'new_cna_shortname') {
+        newCnaShortname = req.query.new_cna_shortname.replace(/["']/g, '')
+        newUser.cna_short_name = newCnaShortname
+      } else if (key === 'active') {
+        active = req.query.active.replace(/["']/g, '').toLowerCase()
+
+        if (active !== 'true' && active !== 'false') {
+          returned = true
+          return res.status(400).json({ message: 'Bad query parameter.' })
+        }
+
+        newUser.active = active
+      } else if (key === 'name.first') {
+        name.first = req.query['name.first'].replace(/["']/g, '')
+        newUser.name.first = name.first
+      } else if (key === 'name.last') {
+        name.last = req.query['name.last'].replace(/["']/g, '')
+        newUser.name.last = name.last
+      } else if (key === 'name.middle') {
+        name.middle = req.query['name.middle'].replace(/["']/g, '')
+        newUser.name.middle = name.middle
+      } else if (key === 'name.suffix') {
+        name.suffix = req.query['name.suffix'].replace(/["']/g, '')
+        newUser.name.suffix = name.suffix
+      } else if (key === 'name.surname') {
+        name.surname = req.query['name.surname'].replace(/["']/g, '')
+        newUser.name.surname = name.surname
+      } else {
+        returned = true
+        return res.status(400).json({ message: 'Invalid query parameter \'' + key + '\'' })
+      }
+    }
+  })
+
+  if (!returned) {
+    User.findOneAndUpdate()
+      .byUserNameAndCnaShortName(username, shortName)
+      .updateOne(newUser)
+      .exec((err, result) => {
+        if (err) {
+          logger.error(err.stack)
+          return res.status(500).send(CONSTANTS.SERVER_RESPONSES.Internal_Server_Error)
+        }
+
+        if (result.n === 0) {
+          logger.info('The user could not be updated because ' + username + ' does not exist for ' + shortName + ' CNA')
+          return res.status(404).json({ message: username + ' could not be found.' })
+        }
+
+        const responseMessage = {
+          message: username + ' was successfully updated.',
+          updated: newUser
+        }
+
+        logger.info(JSON.stringify(responseMessage))
+        return res.status(200).json(responseMessage)
+      })
   }
+}
+
+// Resets the user secret
+async function resetSecret (req, res) {
+  const cna = req.header('CVE-API-CNA')
+  const username = req.params.username
+  const cnaShortName = req.params.shortname
+  const newUser = new User()
+
+  if (cna !== process.env.SECRETARIAT) {
+    logger.info('The API secret of ' + username + ' could not be updated because the user\'s secret cannot be updated by other than the secretariat.')
```

Comment:
```suggestion
    logger.info('The API secret of ' + username + ' could not be updated because the user\'s secret cannot be updated by other than the Secretariat.')
```